src/program.cpp

```diff
@@ -261,21 +300,24 @@ std::vector<argument> program::eval(parameter_map params) const
 
     if(trace_level > 0)
     {
-        return generic_eval(*this, ctx, std::move(params), [&](auto& ins, auto f) {
-            ctx.finish();
-            std::cout << "Run instruction: ";
-            this->debug_print(ins);
-            auto result = check_context(f);
-            ctx.finish();
-            if(trace_level > 1 and ins->name().front() != '@' and ins->name() != "load")
-                std::cout << "Ouput: " << result << std::endl;
-            return result;
-        });
+        std::unordered_map<instruction_ref, std::string> names;
+        return generic_eval(
+            *this, ctx, std::move(params), [&](auto& ins, const module* mod, auto f) {
+                ctx.finish();
+                std::cout << "Run instruction: ";
+                mod->debug_print(ins, names);
```

Comment:
How is names update here? Can't we just do `this->debug(ins)` here?

Reply:
If we do not have the name input, instructions of main module cannot be printed out in submodules.

Reply:
What do you mean? Are you saying `this->debug_print(ins)` is broken?